packages/desktop-client/src/components/LoggedInUser.tsx

```diff
@@ -29,54 +35,83 @@ export function LoggedInUser({
   const [loading, setLoading] = useState(true);
   const [menuOpen, setMenuOpen] = useState(false);
   const serverUrl = useServerURL();
-  const triggerRef = useRef(null);
+  const triggerRef = useRef<HTMLButtonElement>(null);
+  const navigate = useNavigate();
+  const [budgetId] = useLocalPref('id');
+  const [cloudFileId] = useLocalPref('cloudFileId');
+  const location = useLocation();
+  const { hasPermission } = useAuth();
+  const [isOwner, setIsOwner] = useState(false);
+  const multiuserEnabled = useMultiuserEnabled();
 
   useEffect(() => {
-    getUserData().then(() => setLoading(false));
-  }, []);
+    if (getUserData) {
+      getUserData().then(() => setLoading(false));
+    }
+  }, [getUserData]);
 
-  async function onChangePassword() {
-    await closeBudget();
-    window.__navigate('/change-password');
-  }
+  useEffect(() => {
+    if (cloudFileId) {
+      send?.('check-file-access', cloudFileId).then(
```

Comment:
I think send should always be available so shouldn't need the ```?```

Reply:
true